HabitTracker.harris-andy/Program.cs

```diff
@@ -0,0 +1,592 @@
+using System.Globalization;
+using Microsoft.Data.Sqlite;
+
+string connectionString = @"Data Source=HabitTracker.db";
+
+using (SqliteConnection connection = new SqliteConnection(connectionString))
+{
+    connection.Open();
+    var tableCmd = connection.CreateCommand();
+
+    tableCmd.CommandText = @"CREATE TABLE IF NOT EXISTS habits (
+        Id INTEGER PRIMARY KEY AUTOINCREMENT,
+        Date TEXT,
+        Hobby TEXT,
+        Units TEXT,
+        Quantity INTEGER
+        )";
+
+    tableCmd.ExecuteNonQuery();
+
+    using (var command = new SqliteCommand("SELECT COUNT(*) FROM habits;", connection))
+    {
+        var count = Convert.ToInt32(command.ExecuteScalar());
+        if (count == 0)
+        {
+            PopulateDatabase();
+        }
+    }
+    connection.Close();
+}
+
+MainMenu();
+
+void MainMenu()
+{
+    Console.Clear();
+    bool closeApp = false;
+    bool withIds = false;
+    while (closeApp == false)
+    {
+        string message =
+            "--------------------------------------------------\n" +
+            "\n\t\tMAIN MENU\n\n" +
+            "\tWhat would you like to do?\n\n" +
+            "\tType 0 to Close Application\n" +
+            "\tType 1 to View All Records\n" +
+            "\tType 2 to Insert Record\n" +
+            "\tType 3 to Delete Record\n" +
+            "\tType 4 to Update Record\n" +
+            "\tType 5 to View A Record Summary\n" +
+            "\tType 6 to Delete All Records :(\n" +
+            "\tType 7 to Add 100 Rows of Fake Data\n" +
+            "--------------------------------------------------\n";
+
+        int inputNumber = -1;
+        while (inputNumber < 0)
+        {
+            inputNumber = validateNumberEntry(message, isMainMenu: true);
+            Console.Clear();
+        }
+
+        switch (inputNumber)
+        {
+            case 0:
+                Console.WriteLine("\nBye!\n");
+                closeApp = true;
+                Environment.Exit(0);
+                break;
+            case 1:
+                GetAllRecords(withIds);
+                break;
+            case 2:
+                Insert();
+                break;
+            case 3:
+                Delete();
+                break;
+            case 4:
+                Update();
+                break;
+            case 5:
+                GetRecordSummary();
+                break;
+            case 6:
+                DeleteTableContents();
+                break;
+            case 7:
+                PopulateDatabase();
+                break;
+            default:
+                Console.Clear();
+                Console.WriteLine("\nInvalid Command. Give me number!");
+                break;
+        }
+    }
+}
+
+void Insert()
+{
+    Console.Clear();
+    string date = GetDateInput();
+    Console.Clear();
+    string hobby = GetHobby();
+    Console.Clear();
+    string units = GetUnitsInput();
+    Console.Clear();
+    int quantity = GetQuantityInput();

```

Comment:
2 part comment. Part 1: At this point you are aware of the units. So you could pass them into `GetQuantityInput(units)`, see Part 2.